core/controllers/domain_objects_validator.py

```
@@ -223,6 +223,60 @@ def validate_change_dict_for_blog_post(
+        sorted(
+            set(answer_dict.keys()) - set(expected_keys) - {'selected_answer'}
+        )
+    )
+    if missing_keys or extra_keys:
+        raise Exception(
+            'Missing keys: %s, Extra keys: %s' % (missing_keys, extra_keys)
+        )
+
+    question_id = answer_dict['question_id']
+    if not isinstance(question_id, str) or not question_id:
+        raise Exception('question_id must be a non-empty string.')
```

Ruff (0.16.1)

[warning] Create your own exception

(TRY002)